src/Umbraco.Cms.Integrations.SEO.Semrush/Api/Management/Controllers/Token/GetAccessTokenController.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Integrations.SEO.Semrush.Configuration;
+using Umbraco.Cms.Integrations.SEO.Semrush.Models.Dtos;
+using Umbraco.Cms.Integrations.SEO.Semrush.Services;
+
+namespace Umbraco.Cms.Integrations.SEO.Semrush.Api.Management.Controllers.Token
+{
+    public class GetAccessTokenController : TokenControllerBase
+    {
+        public GetAccessTokenController(
+            IOptions<SemrushSettings> options,
+            IWebHostEnvironment webHostEnvironment,
+            ISemrushTokenService semrushTokenService,
+            ICacheHelper cacheHelper,
+            TokenBuilder tokenBuilder,
+            SemrushComposer.AuthorizationImplementationFactory authorizationImplementationFactory,
+            IHttpClientFactory httpClientFactory) : base(options, webHostEnvironment, semrushTokenService, cacheHelper, tokenBuilder, authorizationImplementationFactory, httpClientFactory)
+        {
+        }
+
+        [HttpPost("token/get")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> GetAccessToken([FromBody] AuthorizationRequestDto request)
+        {
+            var result = await _authorizationService.GetAccessTokenAsync(request.Code);
+            return result.StartsWith("Error") ? BadRequest(result) : Ok(result);
```

Comment:
Could you update to `InternalServerError` to align with the `ProductResponseType` attribute status code?